Source/ARTRealtime.m

```diff
@@ -567,12 +613,43 @@ - (void)transportReconnectWithHost:(NSString *)host {
     [self.transport connect];
 }
 
-- (void)transportReconnectWithRenewedToken {
-    _renewingToken = true;
+- (void)transportReconnectWithTokenDetails:(ARTTokenDetails *)tokenDetails error:(NSError *)error {
+    if (error && (self.options.authUrl || self.options.authCallback)) {
+        [self.connection setErrorReason:[ARTErrorInfo createWithCode:ARTCodeErrorAuthConfiguredProviderFailure message:error.localizedDescription]];
+        return;
+    }
     [_transport close];
     _transport = [[_transportClass alloc] initWithRest:self.rest options:self.options resumeKey:_transport.resumeKey connectionSerial:_transport.connectionSerial];
     _transport.delegate = self;
-    [_transport connectForcingNewToken:true];
+    [_transport connectWithToken:tokenDetails.token error:error];
+}
+
+- (void)transportReconnectWithRenewedToken {
+    self.auth.delegate = nil;
+    @try {
+        _renewingToken = true;
+        __weak __typeof(self) weakSelf = self;
+        dispatch_block_t work = artDispatchScheduled([ARTDefault realtimeRequestTimeout], ^{
+            [weakSelf onConnectionTimeOut];
+        });
+        [self.auth authorize:nil options:self.options callback:^(ARTTokenDetails *tokenDetails, NSError *error) {
+            // Cancel scheduled work
+            artDispatchCancel(work);
+            // It's still valid?
+            switch ([[weakSelf connection] state]) {
+            case ARTRealtimeClosing:
+            case ARTRealtimeClosed:
+                return;
+            default:
+                break;
+            }
+            [[weakSelf getLogger] debug:__FILE__ line:__LINE__ message:@"R:%p WS:%p authorised: %@ error: %@", weakSelf, [weakSelf getTransport], tokenDetails, error];
+            [weakSelf transportReconnectWithTokenDetails:tokenDetails error:error];
+        }];
+    }
+    @finally {
+        self.auth.delegate = self;
```

Comment:
Actually, your [comment above](https://github.com/ably/ably-ios/pull/519/files/0c86eb8424812f2fa2cbe8fd092b8341fbaacd85#r91504530) is relevant about this case.